`consul-template`:  revert `function_denylist` logic (#12071)
hashicorp/nomad: .changelog/12071.txt

```release-note:bug

> template: Fixes a bug in config parsing that resulted in the default `function_denylist`

Author: ```suggestion
template: Fixes a bug in config loading that resulted in the default `function_denylist`
```

> template: Fixes a bug in config loading that resulted in the default `function_denylist`
> always being appended to whatever the user specified.

Reviewer: ```suggestion
template: Fixed a bug where the default `function_denylist` would be appended to a specified list
```

Just trying to reword to keep these down to 1 line
